frontend/src/pages/facemash.js

```diff
@@ -34,6 +34,10 @@ class FaceMash extends Component<Props> {
       error: false,
       message: ''
     }
+    const { candidates, generateMatchData } = this.props
+    if (candidates == null || candidates.length != 2) {
+      generateMatchData(this.state.results.candidate1, this.state.results.candidate2)
+    }
```

Comment:
Since the results come from `getInitialProps` which is called from the server, a candidate pairing request will happen every time regardless of whether the client already has a saved pairing locally, because the server has no way of knowing whether the client has one.

Reply:
yes, the server generate matches and validate whether a match has been created when a client submits a match pick. Can we move it away from `getIntialProps`?